Sources/Commands/Workspace.swift

```diff
@@ -651,6 +739,8 @@ public class Workspace {
 
         // First, add the root package constraints.
         var constraints = computeRootPackageConstraints(currentManifests.root)
+        // Add constraints from pins.
+        constraints += pinsStore.createConstraints()
```

Comment:
BTW, I think current resolver might end up including the pinned repo if constraints reference it, but it turns out to not actually be needed (it is an unused repo). We should get a test for this behavior (that is disabled for now, or something).